enhancements/dns/externaldns-operator-aws-assume-role.md

```diff
@@ -0,0 +1,301 @@
+
+### Operational Aspects of API Extensions
+
+N/A
+
+#### Failure Modes
+
+- If the role ARN is not configured with the appropriate permissions, ExternalDNS will error and be unable to generate
+  DNS records.
+
+#### Support Procedures
+
+TODO
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation History`.
+
+## Alternatives
+
+### Bumping the API
+
+The appropriate fix for addressing the problem outlined in [The Required Credentials Issue](#the-required-credentials-issue)
+would be as follows:
+1. Update the CRD schema to make the existing `Credentials` field optional.
+2. Create a new version of the API that makes the optional field a pointer (assuming `Credentials` unset and `AssumeRole`
+   set is a valid combination).
+3. Update new API to not allow zero value of `Credentials`.
+4. Make the conversion webhook convert new/nil to old/zero-value and old/zero-value to new/nil, and new/zero-value to
+   old/zero-value.
+
+However, the concern with bumping the API solely for this update is that the process of bumping can be quite
+time-consuming, and it doesn't align the timeline of this enhancement. Additionally, it would be more efficient to
+bundle this fix with other API updates, such as the need to rebase for incorporating an upstream fix related to the
+Infoblox view parameter (https://github.com/kubernetes-sigs/external-dns/pull/3301) at a later point.
+
+### Assume Role Strategy
+
+Assuming a different AWS role could be implemented without ExternalDNS's built in `--aws-assume-role` as demonstrated in
+https://github.com/openshift/external-dns-operator/pull/191 by using kiam, IRSA, or kube2iam. These are alternate
+methods for assuming a different AWS role, which require utilizing third-party software in addition to ExternalDNS
+(kiam, kube2iam) or the STS support (IRSA: bound service account token). Therefore, this method was not chosen.
+
+### External ID
+
+We considered supporting configuration of an [external ID](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html)
+for the ExternalDNS Operator when using an assumed role. ExternalDNS provides support for an external ID via the [`--aws-assume-role-external-id`](https://github.com/kubernetes-sigs/external-dns/blob/22da9f231dbc6faa3a668b507a4a06823a129609/pkg/apis/externaldns/types.go#L460)
+command line argument. AWS suggests the use of an external ID in specific scenarios to mitigate privilege escalation
+such as the [confused deputy problem](https://en.wikipedia.org/wiki/Confused_deputy_problem). However, for this effort,
+we opted not to include this functionality at the moment. TThe reason behind this decision is that it would become
+necessary to incorporate support for external ID throughout all the components that enable shared VPC (including API,
+Install, and Ingress Operator), and there hasn't been a specific customer need for it thus far.
```

Comment:
Why would it be necessary?  Do you mean that adding external id to the ExternalDNS API would require adding it to the other APIs for consistency?